plugins/processors/noise/noise.go

```diff
@@ -0,0 +1,164 @@
+	"fmt"
+	"math"
+	"reflect"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"gonum.org/v1/gonum/stat/distuv"
+)
+
+const (
+	defaultScale     = 1.0
+	defaultMin       = -1.0
+	defaultMax       = 1.0
+	defaultMu        = 0.0
+	defaultNoiseType = "laplacian"
+)
+
+const sampleConfig = `
+  [[processors.noise]]
+    ## Specified the type of the random distribution.
+    ## Can be "laplacian", "gaussian" or "uniform".
+    # type = "laplacian
+
+    ## Center of the distribution.
+    ## Only used for Laplacian and Gaussian distributions.
+    # mu = 0.0
+
+    ## Scale parameter for the Laplacian or Gaussian distribution
+    # scale = 1.0
+
+    ## Upper and lower bound of the Uniform distribution
+    # min = -1.0
+    # max = 1.0
+
+    ## Apply the noise only to numeric fields matching the filter criteria below.
+    ## Excludes takes precedence over includes.
+    # include_fields = []
+    # exclude_fields = []
+`
+
+type Noise struct {
+	Scale         float64         `toml:"scale"`
+	Min           float64         `toml:"min"`
+	Max           float64         `toml:"max"`
+	Mu            float64         `toml:"mu"`
+	IncludeFields []string        `toml:"include_fields"`
+	ExcludeFields []string        `toml:"exclude_fields"`
+	NoiseType     string          `toml:"type"`
+	Log           telegraf.Logger `toml:"-"`
+	generator     distuv.Rander
+	fieldFilter   filter.Filter
+}
+
+func (p *Noise) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *Noise) Description() string {
+	return "Adds noise to numerical fields"
+}
+
+// generates a random noise value depending on the defined probability density
+// function and adds that to the original value. If any integer overflows
+// happen during the calculation, the result is set to MaxInt or 0 (for uint)
+func (p *Noise) addNoise(value interface{}) interface{} {
+	n := p.generator.Rand()
+	switch v := value.(type) {
+	case int:
+	case int8:
+	case int16:
+	case int32:
+	case int64:
+		if v > -1 && v < 1 {
+			return int64(n)
+		}
+		if v > 0 && (n > math.Nextafter(float64(math.MaxInt64), 0) || int64(n) > math.MaxInt64-v) {
+			p.Log.Debug("Int64 overflow, setting value to MaxInt64")
+			return int64(math.MaxInt64)
+		}
+		if v < 0 && (n < math.Nextafter(float64(math.MinInt64), 0) || int64(n) < math.MinInt64-v) {
+			p.Log.Debug("Int64 (negative) overflow, setting value to MinInt64")
+			return int64(math.MinInt64)
+		}
+		return v + int64(n)
+	case uint:
+	case uint8:
+	case uint16:
+	case uint32:
+	case uint64:
+		if v == uint64(0) {
+			return uint64(math.Abs(n))
+		}
+		if n > math.Nextafter(float64(math.MaxUint64), 0) || uint64(n) > math.MaxUint64-v {
+			p.Log.Debug("Uint64 overflow, setting value to MaxUint64")
+			return uint64(math.MaxUint64)
+		}
+		if n < 0 && uint64(-n) > v {
+			p.Log.Debug("Uint64 (negative) overflow, setting value to 0")
+			return uint64(0)
+		}
+		return v + uint64(n)
```

Comment:
I think this is not correct for negative `n`. So either you do a cast chain (float->int64->uint64) or you do
```suggestion
		if n < 0 {
			if uint64(-n) > v {
				p.Log.Debug("Uint64 (negative) overflow, setting value to 0")
				return uint64(0)
			}
			return v - uint64(-n)
		}
		return v + uint64(n)
```